src/commands/addons/auth.js

```diff
@@ -1,38 +1,18 @@
-const { getAddons } = require('netlify/src/addons')
 const Command = require('../../utils/command')
 const openBrowser = require('../../utils/open-browser')
+const { prepareAddonCommand, ADDON_VALIDATION } = require('../../utils/addons/prepare')
 
 class AddonsAuthCommand extends Command {
   async run() {
-    const accessToken = await this.authenticate()
     const { args } = this.parse(AddonsAuthCommand)
-
     const addonName = args.name
+    const { addon } = await prepareAddonCommand({
+      context: this,
+      addonName,
+      validation: ADDON_VALIDATION.EXISTS,
+    })
 
-    const siteId = this.netlify.site.id
-
-    if (!siteId) {
-      this.log('No site id found, please run inside a site folder or `netlify link`')
-      return false
-    }
-
-    const site = await this.netlify.api.getSite({ siteId })
-    const addons = await getAddons(siteId, accessToken)
-
-    if (typeof addons === 'object' && addons.error) {
```

Comment:
This code was removed: is it ok?

Reply:
Since we're using the `js-client` non `ok` status codes will now throw an error and handled as part of the `try/catch`

Reply:
The old `getAddons` only handled `422`:
https://github.com/netlify/js-client/blob/00b3e939ef133a15db0e9ba8e4e873d748fd442b/src/addons.js#L41

Reply:
:+1: